rs/execution_environment/src/scheduler.rs

```diff
@@ -2061,6 +2077,7 @@ fn execute_canisters_on_thread(
                 &mut round_limits,
                 subnet_size,
             );
+            executed_canister_ids.insert(canister_id);
```

Comment:
Also, shouldn't this go into the `if let Some(instructions_used) = instructions_used` block below? We seem to only count a message as executed if some instructions were executed. As is, we may end up with zero messages and non-zero canisters executed.

(Just making sure you're aware, it may make sense to leave things as they are.)

Reply:
Done.